mapper/thin_edge_json/src/json.rs

```diff
@@ -103,20 +103,29 @@ impl ThinEdgeJson {
         value: &JsonValue,
     ) -> Result<DateTime<FixedOffset>, ThinEdgeJsonError> {
         match value {
+            //When timestamp string is long, with more digits in the timestamp subseconds
+            JsonValue::String(str) => {
+                let result = ThinEdgeJson::parse_from_rfc3339(str)?;
+                Ok(result)
+            }
+            //When timestamp string is having less digits in the subseconds
             JsonValue::Short(str) => {
-                let timestamp = str.as_str();
-                //Parse fails if timestamp is not is8601 complaint
-                let result = DateTime::parse_from_rfc3339(&timestamp).map_err(|err| {
-                    ThinEdgeJsonError::InvalidTimestamp {
-                        value: String::from(timestamp),
-                        from: err,
-                    }
-                })?;
+                let result = ThinEdgeJson::parse_from_rfc3339(str)?;

```

Comment:
When multiple enum match-arms need are doing the same action, you can collapse them like this: https://www.reddit.com/r/rust/comments/aeo0gh/match_statements_multiple_armspatterns_to_one/ . For some reason, this handy style is not documented in the rust book.

Reply:
Collapsing will not work because I capture an item in every handle and forward it to the calling function.

Reply:
Since the types are different it does not work. Here is the compiler error
 JsonValue::String(str) |  JsonValue::Short(str)  => {
    |                               ---                      ^^^ expected struct `std::string::String`, found struct `json::short::Short`